app/src/main/java/com/sun/unsplash_02/screen/home/HomeFragment.kt

```diff
@@ -27,14 +38,64 @@ class HomeFragment : BaseFragment(), HomeContract.View {
     override fun onInit() {
         CollectionAdapter().run {
             setOnItemClickListener {
-                Toast.makeText(context, it.id, Toast.LENGTH_SHORT).show()
+                if (isLoading) {
+                    photoAdapter?.stopLoadMore()
+                    isLoading = false
+                }
+                showLoading()
+                if (currentCollectionId != it.id) {
+                    homePresenter.resetCurrentPage()
+                    recyclerPhoto.smoothScrollToPosition(0)
+                }
+                currentCollectionId = it.id
+                selectedType = TYPE_PHOTO_COLLECTION
+                photoAdapter?.clear()
+                homePresenter.loadListImagesByCollection(it.id)
             }
             collectionAdapter = this
             recyclerCollection.adapter = this
         }
+        PhotoAdapter().run {
+            setItemClickListener(this@HomeFragment)
+            photoAdapter = this
+            recyclerPhoto.adapter = this
+        }
         super.onInit()
     }
 
+    override fun onEvent() {
+        recyclerPhoto.run {
+            smoothScrollToPosition(0)
+            addOnScrollListener(object : RecyclerView.OnScrollListener() {
+
+                override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                    super.onScrolled(recyclerView, dx, dy)
+                    val layoutManager = recyclerPhoto.layoutManager as GridLayoutManager
+                    if (!isLoading) {
+                        photoAdapter?.let {
+                            if (layoutManager.findLastCompletelyVisibleItemPosition() ==
+                                photoAdapter!!.getListPhotos().size - 1
```

Comment:
```suggestion
                                photoAdapter.getListPhotos().size - 1
```

Reply:
cái này do để adapter nullable nên nó bắt check ạ

Reply:
dùng let rồi mà e. Thì trong này phải dùng it chứ không gọi lại